lib/puppet/rest/route.rb

```diff
@@ -0,0 +1,15 @@
+module Puppet::Rest
+  class Route
```

Comment:
Why are Route and ServerResolver separate?

Reply:
They may not need to be, now that I've moved to a model where the base URL is configured during client initialization (I was doing it per-request before, until Josh mentioned that might be expensive). Were you imagining something like Route implementing the server and port resolution method?